notifications/src/main/kotlin/com/amazon/opendistroforelasticsearch/commons/notifications/model/NotificationStatus.kt

```diff
@@ -0,0 +1,152 @@
+            var emailRecipientStatus: List<EmailRecipientStatus> = listOf()
+            var deliveryStatus: DeliveryStatus? = null
+
+            XContentParserUtils.ensureExpectedToken(
+                XContentParser.Token.START_OBJECT,
+                parser.currentToken(),
+                parser::getTokenLocation
+            )
+            while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+                val fieldName = parser.currentName()
+                parser.nextToken()
+                when (fieldName) {
+                    CONFIG_NAME_TAG -> configName = parser.text()
+                    CONFIG_ID_TAG -> configId = parser.text()
+                    CONFIG_TYPE_TAG -> configType = valueOf(parser.text(), NotificationConfig.ConfigType.None)
+                    EMAIL_RECIPIENT_STATUS_TAG -> emailRecipientStatus = parser.objectList { EmailRecipientStatus.parse(it) }
+                    STATUS_DETAIL_TAG -> deliveryStatus = DeliveryStatus.parse(parser)
+                    else -> {
+                        parser.skipChildren()
+                        log.info("Unexpected field: $fieldName, while parsing Notification Status")
+                    }
+                }
+            }
+            configName ?: throw IllegalArgumentException("$CONFIG_NAME_TAG field absent")
+            configId ?: throw IllegalArgumentException("$CONFIG_ID_TAG field absent")
+            configType ?: throw IllegalArgumentException("$CONFIG_TYPE_TAG field absent")
+
+            return NotificationStatus(
+                configId,
+                configName,
+                configType,
+                emailRecipientStatus,
+                deliveryStatus
+            )
+        }
+    }
+
+    /**
+     * Constructor used in transport action communication.
+     * @param input StreamInput stream to deserialize data from.
+     */
+    constructor(input: StreamInput) : this(
+        configId = input.readString(),
+        configName = input.readString(),
+        configType = input.readEnum(NotificationConfig.ConfigType::class.java),
+        emailRecipientStatus = input.readList(EmailRecipientStatus.reader),
+        deliveryStatus = input.readOptionalWriteable(DeliveryStatus.reader)
+        )
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun writeTo(output: StreamOutput) {
+        output.writeString(configId)
+        output.writeString(configName)
+        output.writeEnum(configType)
+        output.writeCollection(emailRecipientStatus)
+        output.writeOptionalWriteable(deliveryStatus)
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun toXContent(builder: XContentBuilder?, params: ToXContent.Params?): XContentBuilder {
+        builder!!
+        return builder.startObject()
+            .field(CONFIG_ID_TAG, configId)
+            .field(CONFIG_TYPE_TAG, configType)
+            .field(CONFIG_NAME_TAG, configName)
+            .fieldIfNotNull(EMAIL_RECIPIENT_STATUS_TAG, emailRecipientStatus)
```

Comment:
emailRecipientStatus is non nullable field so may be we can use field()

Reply:
will fix before merging